fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/query/SelectRowSource.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * (C) Copyright IBM Corp. 2019,2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.database.utils.query;
+
+import com.ibm.fhir.database.utils.query.expression.StatementRenderer;
+import com.ibm.fhir.database.utils.query.expression.StringStatementRenderer;
+
+/**
+ * Represents a table referenced in the from list
+ */
+public class SelectRowSource implements RowSource {
+
+    // The sub-query
+    private final Select select;
+
+    /**
+     * Protected constructor
+     *
+     * @param subQuery
+     * @param alias
+     */
+    protected SelectRowSource(Select subQuery) {
```

Comment:
I noticed this before, I like the protected nature of this

Reply:
ok